code/modules/paperwork/clipboard.dm

```diff
@@ -1,154 +1,157 @@
+	var/obj/item/weapon/pen
+	var/obj/item/weapon/toppaper
 	slot_flags = SLOT_BELT
 	burn_state = FLAMMABLE
 
 /obj/item/weapon/clipboard/New()
+	..()
 	update_icon()
 
-/obj/item/weapon/clipboard/MouseDrop(obj/over_object as obj) //Quick clipboard fix. -Agouri
-	if(ishuman(usr))
-		var/mob/M = usr
-		if(!(istype(over_object, /obj/screen) ))
-			return ..()
-
-		if(!M.restrained() && !M.stat)
-			switch(over_object.name)
-				if("r_hand")
-					M.unEquip(src)
-					M.put_in_r_hand(src)
-				if("l_hand")
-					M.unEquip(src)
-					M.put_in_l_hand(src)
-
-			add_fingerprint(usr)
+/obj/item/weapon/clipboard/verb/removePen()
+	set category = "Object"
+	set name = "Remove clipboard pen"
+	if(!ishuman(usr) || usr.incapacitated())
+		return
+	penPlacement(FALSE, pen)
+
+/obj/item/weapon/clipboard/proc/isPaperwork(var/obj/item/weapon/W) //This could probably do with being somewhere else but for now it's fine here.
+	if(istype(W, /obj/item/weapon/paper) || istype(W, /obj/item/weapon/paper_bundle))
+		return PAPERWORK
+	if(istype(W, /obj/item/weapon/photo))
+		return PHOTO
+
+/obj/item/weapon/clipboard/proc/isPen(var/obj/item/weapon/W) //Should also probably be somewhere else
+	if(istype(W, /obj/item/weapon/pen))
+		return TRUE
+
+/obj/item/weapon/clipboard/proc/checkTopPaper()
+	if(toppaper.loc != src) //Oh no! We're missing a top sheet! Better get another one to be at the top.
+		var/obj/item/weapon/paper/newtoppaper = locate(/obj/item/weapon/paper) in src
+		if(!newtoppaper) //In case there's no paper, try find a paper bundle instead (why is paper_bundle not a subtype of paper?)
+			newtoppaper = locate(/obj/item/weapon/paper_bundle) in src
+		if(newtoppaper)
+			toppaper = newtoppaper
+		else
+			toppaper = null
+
+obj/item/weapon/clipboard/proc/penPlacement(var/placing, var/obj/item/weapon/pen/P)
+	if(placing == TRUE)
+		if(pen)
+			to_chat(usr, "<span class = 'warning'>There's already a pen in [src]!</span>")
+			return
+		if(!isPen(P))
 			return
+		to_chat(usr, "<span class='notice'>You slide [P] into [src].</span>")
+		usr.unEquip(P)
+		P.forceMove(src)
+		pen = P
+	else
+		if(!pen)
+			to_chat(usr, "<span class = 'warning'>There isn't a pen in [src] for you to remove!</span>")
+			return
+		to_chat(usr, "<span class = 'notice'>You remove [pen] from [src].</span>")
+		usr.put_in_hands(pen)
+		pen = null
+	update_icon()
+
+/obj/item/weapon/clipboard/proc/show_clipboard() //Show them what's on the clipboard
+	var/dat = "<title>[src]</title>"
+	dat += "<a href='?src=[UID()];doPenThings=[pen ? "Remove" : "Add"]'>[pen ? "Remove pen" : "Add pen"]</a><br><hr>"
+	for(var/obj/item/weapon/P in src)
+		if(isPaperwork(P) == PAPERWORK)
+			dat += "<a href='?src=[UID()];remove=\ref[P]'>Remove</a> <a href='?src=[UID()];topPaper=\ref[P]'>[toppaper == P ? "On top" : "Put on top"]</a> <a href='?src=[UID()];viewOrWrite=\ref[P]'>[toppaper == P ? "<strong>" : null][P.name][toppaper == P ? "</strong>" : null]</a><br><br>"
```

Comment:
If someone could suggest a better way to do the `<strong></strong>` tags, that'd be grand